pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -389,6 +411,25 @@ func (r *githubRepository) History(ctx context.Context, path, ref string) ([]pro
 	return ret, nil
 }
 
+func (r *githubRepository) isWorkflowAllowed(ref string) error {
+	cfg := r.config.Spec
+
+	workflows := make(map[provisioning.Workflow]bool)
+	for _, w := range cfg.GitHub.Workflows {
+		workflows[w] = true
+	}
+
+	if ref == cfg.GitHub.Branch && !workflows[provisioning.PushWorkflow] {
+		return apierrors.NewBadRequest("direct push not allowed")
+	}
+
+	if ref != cfg.GitHub.Branch && !workflows[provisioning.BranchWorkflow] {
```

Comment:
🤔  is there a PR workflow that does not start with making a branch?

Reply:
addressed